Index: SUSYTools/SUSYTools/SUSYObjDef_xAOD.h
===================================================================
--- SUSYTools/SUSYTools/SUSYObjDef_xAOD.h	(revision 640074)
+++ SUSYTools/SUSYTools/SUSYObjDef_xAOD.h	(working copy)
@@ -136,12 +136,18 @@
 		float GetSignalMuonSF(const xAOD::Muon& mu);
 		
 		float GetSignalElecSF(const xAOD::Electron& el,
-													const bool recoSF = true, const bool idSF = true, const bool triggerSF = true);
+				      const bool recoSF = true, const bool idSF = true, const bool triggerSF = true);
+	  
+                float GetSignalPhotonSF(const xAOD::Photon& ph,
+					const bool recoSF = true, const bool idSF = true, const bool triggerSF = true);
 		
 		float GetSignalTauSF(const xAOD::TauJet& tau);
 		
 		StatusCode OverlapRemoval(const xAOD::ElectronContainer *electrons, const xAOD::MuonContainer *muons, const xAOD::JetContainer *jets, const bool doHarmonization = true, const double dRejet = 0.2, const double dRjetmu = 0.4, const double dRjete = 0.4);
 		
+	  //including photons //MARTIN
+	  virtual StatusCode OverlapRemoval(const xAOD::ElectronContainer *electrons, const xAOD::MuonContainer *muons, const xAOD::JetContainer *jets, const xAOD::PhotonContainer *photons, bool doHarmonization = true, double dRejet = 0.2, double dRjetmu = 0.4, double dRjete = 0.4, double dRphlep=0.4, double dRphjet=0.4);
+
 		CP::SystematicCode resetSystematics();
 		
 		CP::SystematicCode applySystematicVariation( const CP::SystematicSet& systConfig );
