Index: Root/SUSYObjDef_xAOD.cxx
===================================================================
--- Root/SUSYObjDef_xAOD.cxx	(revision 650561)
+++ Root/SUSYObjDef_xAOD.cxx	(working copy)
@@ -14,12 +14,12 @@
 
 namespace ST {
 	
-  static SG::AuxElement::Decorator<bool> dec_baseline("baseline");
-  static SG::AuxElement::Decorator<bool> dec_signal("signal");
-  static SG::AuxElement::Decorator<bool> dec_passOR("passOR");
-  static SG::AuxElement::Decorator<bool> dec_bad("bad");
-  static SG::AuxElement::Decorator<bool> dec_bjet("bjet");
-  static SG::AuxElement::Decorator<bool> dec_cosmic("cosmic");
+  static SG::AuxElement::Decorator<char> dec_baseline("baseline");
+  static SG::AuxElement::Decorator<char> dec_signal("signal");
+  static SG::AuxElement::Decorator<char> dec_passOR("passOR");
+  static SG::AuxElement::Decorator<char> dec_bad("bad");
+  static SG::AuxElement::Decorator<char> dec_bjet("bjet");
+  static SG::AuxElement::Decorator<char> dec_cosmic("cosmic");
 	
   static SG::AuxElement::Decorator<float> dec_effscalefact("effscalefact");
 	
@@ -1022,7 +1022,7 @@
 		
     bool isbaseline = m_tauSelTool->accept( input );
 		
-    dec_baseline(input) = isbaseline;
+    dec_baseline(input) = (char)isbaseline;
 		
     // Return gracefully:
     return StatusCode::SUCCESS;
