Index: EventShapeTools/cmt/Makefile.RootCore
===================================================================
--- EventShapeTools/cmt/Makefile.RootCore	(revision 650561)
+++ EventShapeTools/cmt/Makefile.RootCore	(working copy)
@@ -12,7 +12,7 @@
 PACKAGE_LDFLAGS  = 
 PACKAGE_BINFLAGS = 
 PACKAGE_LIBFLAGS = 
-PACKAGE_DEP      = AsgTools PATCore PATInterfaces xAODJet JetInterface xAODEventShape Asg_FastJet
+PACKAGE_DEP      = AsgTools PATCore PATInterfaces xAODJet JetInterface xAODEventShape Asg_FastJet EventShapeInterface
 PACKAGE_TRYDEP   = 
 PACKAGE_CLEAN    = 
 PACKAGE_PEDANTIC = 0
