Index: SUSYTools/Root/SUSYToolsInit.cxx
===================================================================
--- SUSYTools/Root/SUSYToolsInit.cxx	(revision 640074)
+++ SUSYTools/Root/SUSYToolsInit.cxx	(working copy)
@@ -256,6 +256,7 @@
     if(m_metSystTool.empty()) {
       met::METSystematicsTool *metSystTool = new met::METSystematicsTool("METSystTool");
       ATH_CHECK( metSystTool->initialize() );
+      metSystTool->msg().setLevel( MSG::WARNING );
       m_metSystTool = ToolHandle<IMETSystematicsTool>(metSystTool);
     }
     
